### PR TITLE
C001 add cxx sdk

### PR DESCRIPTION
**Pull request by rbuysse**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#
+#   The image is also used by unit and integration tests. Docker is used to
+#   interact with and orchestrate the creation of other docker containers.
+#
+# Build:
+#   $ cd sawtooth-sdk-cxx
+#   $ docker build . -t sawtooth-sdk-cxx-local
+#
+# Run:
+#   $ cd sawtooth-core
```
Comment (dcmiddle) [inline Dockerfile]: s/core/sdk-cxx

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ && mkdir -p /var/log/sawtooth \
+ && mkdir -p /var/lib/sawtooth \
+ && mkdir -p /etc/sawtooth \
+ && mkdir -p /etc/sawtooth/keys
+
+ENV PATH=$PATH:/project/sawtooth-sdk-cxx/bin
+
+ENV VERSION=AUTO_STRICT
+WORKDIR /project/sawtooth-sdk-cxx
+CMD build_cxx
```
Comment (dcmiddle) [inline Dockerfile]: Error:
```
docker run -v $(pwd):/project/sawtooth-sdk-cxx sawtooth-sdk-cxx-local
...
CMake Error at /usr/share/cmake-3.5/Modules/CPackDeb.cmake:577 (message):
  CPackDeb: Debian package requires a package version
Call Stack (most recent call first):
  /usr/share/cmake-3.5/Modules/CPackDeb.cmake:778 (cpack_deb_prepare_package_vars)
  


CPack Error: Error while execution CPackDeb.cmake
CPack Error: Problem compressing the directory
CPack Error: Error when generating package: sawtooth-cxx-sdk
Makefile:94: recipe for target 'package' failed
make: *** [package] Error 1
```

**Comment by dcmiddle** [comment]
@EugeneYYY did you build using cmake directly on the host os in ubuntu? or did you use the docker method(s)?

File: examples/xo_cxx/address_mapper.cpp [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        out << "Address does not contain 70 "
+            << "characters: " << addr.length() << " != 70";
+        throw AddressFormatError(out.str());
+    } else if (!IsHex(addr)) {
+        throw AddressFormatError("Address must contain only " \
+            "lowercase hexadecimal characters");
+    }
+}
+
+// Checks if an address string meets the constraints of an address.
```
Comment (danintel) [inline examples/xo_cxx/address_mapper.cpp]: s/address string/namespace string/

Comment (danintel) [reply examples/xo_cxx/address_mapper.cpp]: This comment was not addressed, so I created new PR #3 to fix this.

**Pull request merged**